### PR TITLE
Clear schedule feature

### LINKED ISSUE #167: Add `clearschedule` command

### PR DESCRIPTION
**Pull request by xavierpok**
Stacked PR on #159, merge ONLY WHEN #159 is merged

Closes #167 

adds new clear schedule, as shown below in screenshots :

**BEFORE COMMAND**
image
**AFTER COMMAND**
image

List of changes/additions : 

- `clearschedule` command, its parser class, and its command class
- `clearschedule` updates the `Person` by replacing it with an instance w/ an empty schedule and updates its `lastmodifieddatetime`
- As a consequence, its parser inherits from `ClockDependentParser`
- execution of the command works when name only is provided, but provides different message (should not happen in normal operation, but in case)

Test cases (all passed) : 

- `equals()`,`toString` for all classes above
- parsing with excess fields (i.e. any fields at all)
- parsing with syntactically incorrect inputs (e.g. non-integer, non-positive integer)
- execution with logically incorrect index
- parsing with correct fields
- parsing with logically incorrect but syntatically correct field (which throws a command failure but manages to parse)
- execution with out-of-bounds field
- execution with only name
- execution with schedule and name
- execution with schedule w/o name
- execution on empty schedule and no name (fails)

### CODE REVIEW

**Comment by xavierpok** [comment]
Still WIP -- Test cases need updating with respect to new changes. in `master`

**Comment by xavierpok** [comment]
Test cases updated, all test cases passed.

Opened for review as #159 got merged.

**Comment by xavierpok** [comment]
Screenshots not updated for latest version of app, but smoke test done on my local machine. 

Functionality still unchanged

File: src/main/java/connexion/logic/commands/ClearScheduleCommand.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+ * Clears the schedule of a contact identified using it's displayed index from the address book.
+ */
+public class ClearScheduleCommand extends Command {
+
+
+    public static final String COMMAND_WORD = "clearschedule";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Removes the schedule of the person identified by the index number used in the displayed person list\n"
+            + "Parameters: INDEX (must be a positive integer) ";
+    public static final String MESSAGE_CLEAR_SCHEDULE_SUCCESS = "Cleared : %1$s";
```
Comment (imkwokyong) [inline src/main/java/connexion/logic/commands/ClearScheduleCommand.java:37]: Will it be better to specify "Clear schedule of :" here?"
"Cleared" here is a bit confusing as it looks like it is clearing the contact
<br>
image

**Review by geoffong11 — APPROVED** [review]
LGTM! Just update the issues raised in another PR.

File: src/main/java/connexion/logic/commands/ClearScheduleCommand.java (hunk 1 above)
Comment (xavierpok) [reply src/main/java/connexion/logic/commands/ClearScheduleCommand.java]: Suggestion added in latest commit

**Pull request merged**